keyboards/xd75/keymaps/buzzlighter1/keymap.c

```diff
@@ -0,0 +1,111 @@
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ * | P*     | P7     | P8     | P9     | +      | NumLk  | PR SCR | SCR LK | PAUSE  | XXXXXXX| XXXXXXX| BL_DEC | BL_INC | RGB HD | RGB HI |
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ * | _______| P4     | P5     | P6     | PENT   | CALC   | XXXXXXX| POWER  | XXXXXXX| RESET  | XXXXXXX| BL_BRTG| BL_STEP| RGB SD | RGB SI |
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ * | P/     | P1     | P2     | P3     | -      | OS     | XXXXXXX| XXXXXXX| XXXXXXX| XXXXXXX| XXXXXXX| BL_OFF | BL_ON  | RGB VD | RGB VI |
+ * |--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ * | CPSLCK | P0     | P.     | _______|                             SPACE                            | BL_TOGG| RGB TG | RGB RMD| RGB MD |
+ * '--------------------------------------------------------------------------------------------------------------------------------------'
+ */
+
+  [_FN] = LAYOUT_ortho_5x15( /* FUNCTION */
+    XXXXXXX, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______,
+    KC_PAST, KC_P7,   KC_P8,   KC_P9,   KC_PPLS, KC_NLCK, KC_PSCR, KC_SLCK, KC_PAUS, XXXXXXX, XXXXXXX, BL_DEC,  BL_INC,  RGB_HUD, RGB_HUI,
+    _______, KC_P4,   KC_P5,   KC_P6,   KC_PENT, KC_CALC, XXXXXXX, KC_PWR,  XXXXXXX, RESET,   XXXXXXX, BL_BRTG, BL_STEP, RGB_SAD, RGB_SAI,
+    KC_PSLS, KC_P1,   KC_P2,   KC_P3,   KC_PMNS, KC_LWIN, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, BL_OFF,  BL_ON,   RGB_VAD, RGB_VAI,
+    KC_CAPS, KC_P0,   KC_PDOT, _______, XXXXXXX, XXXXXXX, XXXXXXX, KC_SPC,  XXXXXXX, XXXXXXX, XXXXXXX, BL_TOGG, RGB_TOG, RGB_RMOD,RGB_MOD
+  )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QMKBEST:
+      if (record->event.pressed) {
+        // when keycode QMKBEST is pressed
+        SEND_STRING("QMK is the best thing ever!");
+      } else {
+        // when keycode QMKBEST is released
+      }
+      break;
+    case QMKURL:
+      if (record->event.pressed) {
+        // when keycode QMKURL is pressed
+        SEND_STRING("https://qmk.fm/" SS_TAP(X_ENTER));
+      } else {
+        // when keycode QMKURL is released
+      }
+      break;
+  }
+  return true;
+}
+
+void matrix_init_user(void) {
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+        capslock_led_on();
+    } else {
+        capslock_led_off();
+    }
+}
+
+// Runs whenever there is a layer state change.
+uint32_t layer_state_set_user(uint32_t state) {
```

Comment:
This should be changed:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

This allows the layer bitmask to be changed in size, if needed/wanted